docs/types.rst

```diff
@@ -726,9 +719,10 @@ Arrays
 ------
 
 Arrays can have a compile-time fixed size or they can be dynamic.
-For storage arrays, the element type can be arbitrary (i.e. also other
-arrays, mappings or structs). For memory arrays, it cannot be a mapping and
-has to be an ABI type if it is an argument of a publicly-visible function.
+The are very few restrictions for the element, it can also be
+another array, a mapping or a struct. The general restrictions for
+types apply, though, in that mappings can only be used in storage
```

Comment:
"… you can only use mappings in storage…"

Reply:
I would like to keep it like that to emphasize that it means that mappings can only be used in storage as opposed to mappings are the only types that can be used in storage.

Reply:
@chriseth Ahh, now I wonder if there may even be better ways to clarify this. Maybe (and feel free to paraphrase):

"You can only store/use mappings in the storage data location" (I believe that's the term we started to use)

Or, keeping it passive, but maybe a bit wordy:

"mappings can only be stored in the storage data location"

Reply:
I would use "storage data location" only for the data location specifier, and not the general term. For example in `contract c { mapping(uint => uint) x; }` the mapping is in storage but there is no explicit data location specifier.

Reply:
OK, not to worry, whatever you feel is best from your initial idea, my ideas, or a combination of.